src/PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.hpp

```diff
@@ -129,4 +129,128 @@ tnsr::ii<DataType, SpatialDim, Frame> extrinsic_curvature(
+    const Scalar<DataType>& lapse,
+    const tnsr::AA<DataType, SpatialDim, Frame>& inverse_spacetime_metric,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi) noexcept;
+
+/*!
+ * \ingroup GeneralRelativityGroup
+ * \brief Computes time derivative of the shift vector from
+ *        the generalized harmonic and geometric variables
+ *
+ * \details Let the generalized harmonic spatial derivative variable be
+ * \f$\phi_{iab} = \partial_i \psi_{ab} \f$. The time derivative of \f$ N^i \f$
+ * is given by:
+ * \f{align*}
+ *  -N^2 n^b \pi_{kb} g^{ik} + N N^j n^b \phi_{jkb} g^{ik}
+ *  =& N g^{ik} n^b (\partial_0 - N^j\partial_j) \psi_{kb}
+ *                  + N g^{ik} n^b N^j\partial_j \psi_{kb} \\
+ *  =& N g^{ik} n^b \partial_0 \psi_{kb} \\
+ *  =& g^{ik} \partial_0 (g_{kj} N^j) - N^j g^{ik} \partial_0 g_{kj} \\
+ *  =& \partial_0 N^i
+ * \f}
+ * where the second to last to last line used the same manipulations
+ * as at the end of the spatial derivative calculation. Therefore:
+ * \f{align*}
+ * \partial_0 N^i =& -N^2 n^b\pi_{kb} g^{ik} + N N^j n^b\phi_{jkb} g^{ik} \\
+ *                =& -N g^{ik} n^b (N \pi_{kb} - N^j \phi_{jkb})
+ * \f}
+ */
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::I<DataType, SpatialDim, Frame> time_deriv_of_shift(
+    const Scalar<DataType>& lapse,
+    const tnsr::I<DataType, SpatialDim, Frame>& shift,
+    const tnsr::II<DataType, SpatialDim, Frame>& inverse_spatial_metric,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    const tnsr::aa<DataType, SpatialDim, Frame>& pi) noexcept;
+
+/*!
+ * \ingroup GeneralRelativityGroup
+ * \brief Computes time derivative of index lowered shift from generalized
+ *        harmonic variables, spatial metric and its time derivative.
+ *
+ * \details The time derivative of \f$ N_i \f$ is given by:
+ * \f{align*}
+ *  \partial_0 N_i = g_{ij} \partial_0 N^j + N^j \partial_0 g_{ij}
+ * \f}
+ * where the first term is obtained from `time_deriv_of_shift`, and the latter
+ * is a user input.
+ */
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::i<DataType, SpatialDim, Frame> time_deriv_of_lower_shift(
+    const Scalar<DataType>& lapse,
+    const tnsr::I<DataType, SpatialDim, Frame>& shift,
+    const tnsr::ii<DataType, SpatialDim, Frame>& spatial_metric,
+    const tnsr::II<DataType, SpatialDim, Frame>& inverse_spatial_metric,
+    const tnsr::ii<DataType, SpatialDim, Frame>& dt_spatial_metric,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    const tnsr::aa<DataType, SpatialDim, Frame>& pi) noexcept;
+
+/*!
+ * \ingroup GeneralRelativityGroup
+ * \brief Computes norm of the shift vector.
+ *
+ * \details Compute norm of the shift vector: \f$ N^i N_i = g_{ij} N^i N^j \f$
```

Comment:
Question for my own understanding, do we use norm to refer to |N| or |N|^2? Just making sure the formula matches with the wording :+1:

Reply:
I used it to mean a norm in an L2 sense with the metric thrown in (`=N^i N_i`).